To-Do-app/src/Components/TodoTaskList.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { Dispatch, SetStateAction } from 'react'
+import { Grid } from '@material-ui/core'
+import { ITask } from '../Interface'
+import TodoTask from './TodoTask'
+interface Props {
+    todoList: ITask[]
+    setTodoList: Dispatch<SetStateAction<ITask[]>>
+}
+
+function TodoTaskList({ todoList, setTodoList }: Props) {
+
+    const deleteTask = (taskNameToDelete: number): void => {
+        setTodoList(todoList.filter((task) => {
+            return task.id !== taskNameToDelete
+        }))
+    }
+
+    const editTask = (taskEditId: number, taskEditName: string, taskEditDescription: string): void => {
+        console.log('taskEditName', taskEditName)
+        setTodoList((todoList) =>
+        (todoList.map((task) => {
```

Comment:
why do you need to use a map here?

Reply:
task is passed as a callback to map and It is used to return the updated array